### PR TITLE
Move version ref from config file to command line flag

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/11124

The goal here is to simplify setting a specific version of apiextensions to parse, moving the parameter from the config file to a command line flag.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by ubergesundheit** [comment]
the `--version` flag is usually used to make the program print its version and exit. I suggest changing this to `--commit-reference`

**Comment by marians** [comment]
ACK, renamed

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Breaking: Remove configuration option `commit_reference`, add command line flag `--version` for the same purpose instead.
```
Comment (stone-z) [inline CHANGELOG.md:10]: Flag has been renamed

Comment (marians) [reply CHANGELOG.md]: Thanks, solved

File: main.go [hunk 2]
```diff
@@ -38,6 +38,9 @@ type CRDDocsGenerator struct {
 
 	// Path to the CRD page template file
 	templateFilePath string
+
+	// git reference (tag, commit SHA, branch name) to check out for thee source repository
```
Comment (stone-z) [inline main.go:42]: nit
```suggestion
	// git reference (tag, commit SHA, branch name) to check out for the source repository
```

**Review by stone-z — APPROVED** [review]
This is reasonable to me. I think the usage is slightly awkward with config + cli flag required but for the intended usage it is fine

**Comment by marians** [comment]
Thanks! The `--commit-reference` will become optional next.

**Pull request merged**